### PR TITLE
Make separate DROP_PART log entry type

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

It's supposed to make the code more readable and straightforward because `DROP_RANGE` and `DROP_PART` are processed in a bit different ways (in particular, `DROP_PART` does not have virtual parts)

### CODE REVIEW

**Review by alesapin — APPROVED** [review]

**Comment by alesapin** [comment]
good code simplification LGTM

File: src/Storages/MergeTree/DropPartsRanges.h [hunk 1]
```diff
@@ -23,20 +23,20 @@ class DropPartsRanges
...
-    bool hasDropRange(const MergeTreePartInfo & new_drop_range_info, MergeTreePartInfo * out_drop_range_info = nullptr) const;
+    bool hasDropPart(const MergeTreePartInfo & new_drop_range_info, MergeTreePartInfo * out_drop_range_info = nullptr) const;
 
     /// Add DROP_RANGE to map
-    void addDropRange(const ReplicatedMergeTreeLogEntryPtr & entry);
+    void addDropPart(const ReplicatedMergeTreeLogEntryPtr & entry);
 
     /// Remove DROP_RANGE from map
-    void removeDropRange(const ReplicatedMergeTreeLogEntryPtr & entry);
+    void removeDropPart(const ReplicatedMergeTreeLogEntryPtr & entry);
```
[the remark this answers is not in the document]
Comment (tavplubix) [reply src/Storages/MergeTree/DropPartsRanges.h:39]: `DropPartsRanges` already sounds to me like something about DROP_PART :)

**Comment by alesapin** [comment]
Performance -- unrelated, flaky.

**Comment by alesapin** [comment]
Rabbit MQ should be already fixed https://github.com/ClickHouse/ClickHouse/issues/45160

**Pull request merged**